### DISCUSSION TITLE
Data structure

### DISCUSSION THREAD

**Discussion by dgkeyes** [rfortherestofus/api-food-asset-map#39] [General]
Hi @grahamammal and @katiejolly! I've been thinking about the data structure for this. Right now we create one object for each of the categories: 

- Providers that offer food pantries
- Providers that offer free, prepared food/hot meals
- Food banks
- Food pharmacies
- Drug stores
- Ethnic food markets
- Full service grocery stores/supermarkets/markets (non SNAP or WIC)
- Stores that accept SNAP/WIC (full service grocery stores, supermarkets, markets, ethnic food markets, Target, Walmart)
- Corner stores
- Fast food restaurants
- Restaurants
- Farmers markets
- Liquor Stores

I'm wondering if we should just create a single data frame that holds everything? A couple reasons why:

1. It would simplify things if we need to have the client do any manual checking on things like tagging ethnic food markets. 
2. It would also make the merging of the SNAP/WIC data much simpler. Because the stores in the SNAP/WIC data are in multiple categories, we're going to have to do multiple merges (e.g. merge SNAP/WIC data with grocery stores, then with drug stores, etc), making it more complicated. If we put everything in a single data frame with a category variable, we can do just one merge. 

Do either of you see any downsides to doing this? I was thinking about whether it would be work with the leaflet portion of the code. For example, right now we bring in data one piece at a time. For example:

```
  addCircles(data = food_pantries,
             radius = 3,
             group = "Food pantries",
             color = gray,
             opacity = .8,
             label = pantry_labels,
             labelOptions = labelOptions(
               style = list("font-weight" = "normal", padding = "3px 8px", "color" = gray),
               textsize = "15px",
               direction = "auto"))
```

However, if we have one large data frame called, say, food_assets, we could just do this: 

```
  addCircles(data = filter(food_assets, category == "Food Pantries"),
             radius = 3,
             group = "Food pantries",
             color = gray,
             opacity = .8,
             label = pantry_labels,
             labelOptions = labelOptions(
               style = list("font-weight" = "normal", padding = "3px 8px", "color" = gray),
               textsize = "15px",
               direction = "auto"))
```

Let me know if you have any thoughts on this. Unless there are major objections or something I'm completely missing, I think we should go this direction.

**Comment by katiejolly** [comment]
Yep, that works!

**Reply by dgkeyes** [reply]
Super! @grahamammal if this all sounds good to you, can you please work on creating one giant data frame with all food resources?

**Reply by grahamammal** [reply]
Sounds great, I'll get started on this right now

**Comment by katiejolly** [comment]
One additional thing that would be helpful as we are creating the final
dataset is to add a more human readable label for the categories. You'll
notice in the script I use for reading in the data for the map I create
different category labels: capitalized, correct spacing, etc. I can also
add them, but if Ellen wouldn't mind then that would be great :)

**Reply by grahamammal** [reply]
Sure I can do that when making the big data set, I'll just use lightly edited versions of the labels in the readme if that works!

**Reply by dgkeyes** [reply]
Perfect, thanks to both of you!

**Reply by dgkeyes** [reply]
Also, Ellen, I'll wait on doing any reverse geocoding until after we have one big data frame. It will be easier at that point.

**Reply by grahamammal** [reply]
Katie I added some category names in 2f8dae2f0fc940ce3695dcc1cb5d227c25070dd1, feel free to change any to better fit on the map or if I misinterpreted one of the osm data files. Also, the encoding issue is now fixed on my end, you should make sure it's all working on yours as well

**Reply by dgkeyes** [reply]
Thanks @katiejolly and @grahamammal for all of your work this weekend! I normally work on Sunday mornings, but didn't today due to Mother's day. I'll take a look at everything tomorrow and get back to you then. Thanks again!
